File: src/types/models/disks.ts
export interface DiskUsage {
  max_file_size?: number;
  max_path_length?: number;
  usage?: number;
}

export interface DiskPartition {
  device: string;
  mount_point: string;
  filesystem_type: string;
  options: string;
  max_file_size: number;
  max_path_length: number;
  usage?: DiskUsage;
}

export interface DiskIOCounters {
  read_count: number;
  write_count: number;
  read_bytes: number;
  write_bytes: number;
  read_time: number;
  write_time: number;
}

export interface Disk {
  name: string;
  partitions: DiskPartition[];
  io_counters?: DiskIOCounters;
}

export interface Disks {
  devices: Disk[];
  io_counters?: DiskIOCounters;
}

File: backend/data/module/disks.go
package data_module

import "github.com/charmbracelet/log"

// DiskIOCounters represents disk I/O statistics
type DiskIOCounters struct {
	ReadCount  int64 `json:"read_count"`
	WriteCount int64 `json:"write_count"`
	ReadBytes  int64 `json:"read_bytes"`
	WriteBytes int64 `json:"write_bytes"`
	ReadTime   int64 `json:"read_time"`
	WriteTime  int64 `json:"write_time"`
}

// DiskUsage represents disk space usage information
type DiskUsage struct {
	Total   int64   `json:"total"`
	Used    int64   `json:"used"`
	Free    int64   `json:"free"`
	Percent float64 `json:"percent"`
}

// DiskPartition represents information about a disk partition
type DiskPartition struct {
	Device         string     `json:"device"`
	MountPoint     string     `json:"mount_point"`
	FilesystemType string     `json:"filesystem_type"`
	Options        string     `json:"options"`
	MaxFileSize    int64      `json:"max_file_size"`
	MaxPathLength  int64      `json:"max_path_length"`
	Usage          *DiskUsage `json:"usage"`
}

// Disk represents information about a single disk device
type Disk struct {
	Name       string          `json:"name"`
	Partitions []DiskPartition `json:"partitions"`
	IOCounters *DiskIOCounters `json:"io_counters"`
}

// DisksData represents information about all disk devices
type DisksData struct {
	Devices    []Disk          `json:"devices"`
	IOCounters *DiskIOCounters `json:"io_counters"`
}

func (t *Module) UpdateDisksModule() (DisksData, error) {
	log.Info("Getting disks data")

	var disksData DisksData
	// Initialize arrays
	disksData.Devices = make([]Disk, 0)

	// TODO: Implement
	return disksData, nil
}
